### PR TITLE
Scritps: Fix load backup and add create backup

### PR DESCRIPTION
**Pull request by flomonster**

### CODE REVIEW

File: scripts/load-backup.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+echo "Note: You should only have your postgres container runnning!"
+
+# Copy needed files to the container
+docker cp "$1" osrd-postgres:/tmp/backup-osrd
+
+# Drop database
+echo Deleting osrd database...
+docker exec osrd-postgres psql -c 'drop database osrd;' > /dev/null
+
+echo Initialzie new database...
```
Comment (multun) [inline scripts/load-backup.sh]: little typo

File: scripts/load-backup.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+docker exec osrd-postgres psql -f /docker-entrypoint-initdb.d/init.sql > /dev/null
+
+# restoring the backend can partialy fail, and that's sometimes ok
+echo Restore backup...
```
Comment (multun) [inline scripts/load-backup.sh:71]: I somewhat assumed this script would validate the backup using the hash in the filename
It does not seem to, is it intentional?

Comment (flomonster) [reply scripts/load-backup.sh:71]: I added the check

File: scripts/load-backup.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+# Check sha1 is matching
+echo Checking backup integrity...
+BACKUP_PATH="$1"
+EXPECTED_SHA1=$(basename -s ".backup" "$BACKUP_PATH")
+EXPECTED_SHA1=${EXPECTED_SHA1##*sha1_}
+CURRENT_SHA1=$(cut -d' ' -f1 <<< $(sha1sum "$BACKUP_PATH"))
+if [ "$EXPECTED_SHA1" == "$CURRENT_SHA1" ]; then
+  echo "  ✔ The backup is valid"
+else
```
Comment (multun) [inline scripts/load-backup.sh]: It does not work if the file was renamed after download.
Failing in this case is ok, but we should detect that the file was renamed and print something

File: scripts/load-backup.sh [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    echo "  You must only have your postgres container runnning!"
+    exit 2
+  fi
+
+  # Drop database
+  echo Deleting osrd database...
+  docker exec osrd-postgres psql -c 'DROP DATABASE osrd;' > /dev/null
+fi
+
+echo Initialze new database...
```
Comment (multun) [inline scripts/load-backup.sh]: typo in Initialze

File: scripts/load-backup.sh [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  echo "usage: $0 osrd.backup"
+  exit 1
+fi
+
+# Check sha1 is matching
+echo Checking backup integrity...
+BACKUP_PATH="$1"
+BACKUP_FILENAME=$(basename -s ".backup" "$BACKUP_PATH")
+EXPECTED_SHA1=$(grep -o -E '[0-9a-f]{40}' <<< $(echo "$BACKUP_FILENAME") || echo "renamed")
+CURRENT_SHA1=$(cut -d' ' -f1 <<< $(sha1sum "$BACKUP_PATH"))
```
Comment (multun) [inline scripts/load-backup.sh]: `<<<` only works if /bin/sh is bash
```suggestion
EXPECTED_SHA1=$(echo "$BACKUP_FILENAME" | grep -o -E '[0-9a-f]{40}' || echo "renamed")
CURRENT_SHA1=$(sha1sum "$BACKUP_PATH" | cut -d' ' -f1)
```
there's the same issue in other places

File: scripts/load-backup.sh [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  echo "  ✘ Corrupted backup (you should download it again)"
+  echo "    Expected sha1: ${EXPECTED_SHA1}"
+  echo "    Current sha1:  ${CURRENT_SHA1}"
+  exit 2
+fi
+
+# Check if the database exists
+echo "Checking database exists..."
+DB_EXISTS=$(docker exec osrd-postgres psql -c "SELECT EXISTS (SELECT FROM pg_stat_database WHERE datname = 'osrd');")
+DB_EXISTS=$(grep -o -E '[tf]$' <<< $(echo "$DB_EXISTS"))
```
Comment (Khoyo) [inline scripts/load-backup.sh]: There is probably some quotes missing around here

```suggestion
DB_EXISTS="$(docker exec osrd-postgres psql -c "SELECT EXISTS (SELECT FROM pg_stat_database WHERE datname = 'osrd');")"
DB_EXISTS="$(echo "$DB_EXISTS" | grep -o -E '[tf]$')"
```

File: scripts/load-backup.sh [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  echo "  Database 'osrd' found"
+else
+  echo "  Database 'osrd' not found"
+fi
+
+if [ $DB_EXISTS == 't' ]; then
+  # Check that no service is connected to the database
+  echo "Checking database availability..."
+  DB_CONN=$(docker exec osrd-postgres psql -c "SELECT numbackends FROM pg_stat_database WHERE datname = 'osrd';")
+  DB_CONN=$(grep -o -E '[0-9]+$' <<< $(echo "$DB_CONN"))
```
Comment (Khoyo) [inline scripts/load-backup.sh]: There are probably some quotes missing around here

```suggestion
  DB_CONN="$(docker exec osrd-postgres psql -c "SELECT numbackends FROM pg_stat_database WHERE datname = 'osrd';")"
  DB_CONN="$(echo "$DB_CONN" | grep -o -E '[0-9]+$')"
```

**Review by Khoyo — COMMENTED** [review]
Otherwise lgtm

File: scripts/create-backup.sh [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+fi
+
+# Generating the backup
+echo Generating the backup
+docker exec osrd-postgres pg_dump -d osrd -F c -Z 9 -f /tmp/osrd.backup > /dev/null
+
+# Get metadata
+echo Collecting backup information
+SHA1=$(cut -d' ' -f1 <<< $(docker exec osrd-postgres sha1sum /tmp/osrd.backup))
+SIZE=$(cut -f1 <<< $(docker exec osrd-postgres du -sh /tmp/osrd.backup))
```
Comment (Khoyo) [inline scripts/create-backup.sh]:
```suggestion
SHA1="$(docker exec osrd-postgres sha1sum /tmp/osrd.backup | cut -d' ' -f1)"
SIZE="$(docker exec osrd-postgres du -sh /tmp/osrd.backup | cut -f1)"
```

File: scripts/create-backup.sh [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+fi
+
+# Generating the backup
+echo "Generating the backup... (it can take some time)"
+docker exec osrd-postgres pg_dump -d osrd -F c -Z 9 -f /tmp/osrd.backup > /dev/null
+
+# Get metadata
+echo Collecting backup information...
+SHA1=$(echo $(docker exec osrd-postgres sha1sum /tmp/osrd.backup) | cut -d' ' -f1)
+SIZE=$(echo $(docker exec osrd-postgres du -sh /tmp/osrd.backup) | cut -f1)
```
Comment (multun) [inline scripts/create-backup.sh]:
```suggestion
SHA1=$(docker exec osrd-postgres sha1sum /tmp/osrd.backup | cut -d' ' -f1)
SIZE=$(docker exec osrd-postgres du -sh /tmp/osrd.backup | cut -f1)
```

File: scripts/load-backup.sh [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+  exit 1
+fi
+
+# Check sha1 is matching
+echo Checking backup integrity...
+BACKUP_PATH="$1"
+BACKUP_FILENAME=$(basename -s ".backup" "$BACKUP_PATH")
+echo "$BACKUP_FILENAME"
+EXPECTED_SHA1=$(echo "$BACKUP_FILENAME" | grep -o -E '[0-9a-f]{40}' || echo "renamed")
+CURRENT_SHA1=$(echo $(sha1sum "$BACKUP_PATH") | cut -d' ' -f1)
```
Comment (multun) [inline scripts/load-backup.sh]:
```suggestion
CURRENT_SHA1=$(sha1sum "$BACKUP_PATH" | cut -d' ' -f1)
```

**Review by multun — APPROVED** [review]

**Review by Khoyo — APPROVED** [review]

**Pull request merged**